instaclustr/resource_azure_peering.go

```diff
@@ -0,0 +1,178 @@
+				Required: true,
+			},
+
+			"peer_suscription_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_vpc_net": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"suscription_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"peer_subnet": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnets"},
+			},
+
+			"peer_subnets": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnet"},
+			},
+
+			"v_net": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func AzureresourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+
+	cdcID, err := VpcPeeringCreate(d, meta)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating VPC peering request object: %s", err)
+	}
+	var createData CreateAzureVPCPeeringRequest
+	createData, err = AzurecreateVpcPeeringRequest(d)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+
+	var jsonStr []byte
+	jsonStr, err = json.Marshal(createData)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+	var id string
+	id, err = client.CreateVpcPeering(cdcID, jsonStr)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in creating the cluster: %s", err)
+	}
+	d.SetId(id)
+	d.Set("peer_suscription_id", id)
+	d.Set("cdc_id", cdcID)
+	var vpcPeering *AzureVPCPeering
+	vpcPeering, err = client.AzureReadVpcPeering(cdcID, id)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in reading GCP VPC peering connection: %s", err)
+	}
+
+	d.Set("peer_suscription_id", vpcPeering.PeerSubscriptionId)
+
+	log.Printf("[INFO] VPC peering request %s has been created.", id)
+	return nil
+}
+
+func AzureresourceVpcPeeringRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+	cluster, err := client.ReadCluster(d.Get("cluster_id").(string))
+	if err != nil {
+		return fmt.Errorf("[Error] Error retrieving cluster info: %s", err)
+	}
+	cdcID := cluster.DataCentres[0].ID
+	vpcPeeringID := d.Get("peer_suscription_id").(string)
+
+	log.Printf("[INFO] Reading the status of VPC peering connection %s.", vpcPeeringID)
+	vpcPeering, err := client.AzureReadVpcPeering(cdcID, vpcPeeringID)
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading VPC peering connection: %s", err)
+	}
+	err = MapAzureVPCPeeringToResource(d, vpcPeering)
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading VPC peering connection: %s", err)
+	}
+	return nil
+
+}
+
+func MapAzureVPCPeeringToResource(d *schema.ResourceData, vpcPeering *AzureVPCPeering) error {
+	if vpcPeering.ID == "" {
+		return nil
```

Comment:
Surely if the given `vpcPeering` ID is null this should return an error right? Why do we return `nil` instead?